packages/controllers/src/cronjob/CronjobController.test.ts

```diff
@@ -0,0 +1,370 @@
+  });
+
+  it('executes cronjobs that were missed during daily check in', async () => {
+    const rootMessenger = getRootCronjobControllerMessenger();
+    const controllerMessenger =
+      getRestrictedCronjobControllerMessenger(rootMessenger);
+
+    const callActionMock = jest
+      .spyOn(controllerMessenger, 'call')
+      .mockImplementation((method) => {
+        if (method === 'SnapController:getAll') {
+          return [getTruncatedSnap()];
+        } else if (method === 'PermissionController:getPermissions') {
+          return { [SnapEndowments.Cronjob]: MOCK_CRONJOB_PERMISSION } as any;
+        }
+        return false;
+      });
+
+    const cronjobController = new CronjobController({
+      messenger: controllerMessenger,
+    });
+
+    // Update state manually for test
+    // eslint-disable-next-line dot-notation
+    cronjobController['update'](() => {
+      return {
+        jobs: {
+          [`${MOCK_SNAP_ID}-0`]: { lastRun: 0 },
+        },
+      };
+    });
+
+    await cronjobController.dailyCheckin();
+
+    jest.advanceTimersByTime(inMilliseconds(24, Duration.Hour));
+
+    expect(callActionMock).toHaveBeenNthCalledWith(
+      5,
+      'SnapController:handleRequest',
+      {
+        snapId: MOCK_SNAP_ID,
+        origin: '',
+        handler: HandlerType.OnCronjob,
+        request: {
+          method: 'exampleMethodOne',
+          params: ['p1'],
+        },
+      },
+    );
+
+    cronjobController.destroy();
+  });
+
+  it('does not schedule cronjob that is too far in the future', async () => {
+    // Ensure that Cronjob will not yet be scheduled if it reaches DAILY_TIMEOUT
+    const MOCK_TOO_FAR_CRONJOB_PERMISSION = deepClone(MOCK_CRONJOB_PERMISSION);
+    MOCK_TOO_FAR_CRONJOB_PERMISSION.caveats[0].value = {
+      jobs: [
+        {
+          expression: '59 23 29 2 *',
+          request: {
+            method: 'exampleMethodOne',
+            params: ['p1'],
+          },
+        },
+      ],
+    };
+    const rootMessenger = getRootCronjobControllerMessenger();
+    const controllerMessenger =
+      getRestrictedCronjobControllerMessenger(rootMessenger);
+
+    const callActionMock = jest
+      .spyOn(controllerMessenger, 'call')
+      .mockImplementation((method) => {
+        if (method === 'SnapController:getAll') {
+          return [getTruncatedSnap()];
+        } else if (method === 'PermissionController:getPermissions') {
+          return {
+            [SnapEndowments.Cronjob]: MOCK_TOO_FAR_CRONJOB_PERMISSION,
+          } as any;
+        }
+        return false;
+      });
+
+    const cronjobController = new CronjobController({
+      messenger: controllerMessenger,
+    });
+
+    await cronjobController.register(MOCK_SNAP_ID);
+
+    expect(callActionMock).toHaveBeenCalledWith(
+      'PermissionController:getPermissions',
+      MOCK_SNAP_ID,
+    );
+
+    // eslint-disable-next-line dot-notation
+    expect(cronjobController['_timers'].size).toBe(0);
```

Comment:
```suggestion
    expect(cronjobController._timers.size).toBe(0);
```

Reply:
This is changed now to use `getter`: `expect(cronjobController.timers.size).toBe(0);`.